### PR TITLE
CHE-1770: New layout for forms in UD

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

Provides new layout for form in UD.
### What issues does this PR fix or reference?

[https://github.com/eclipse/che/issues/1770](https://github.com/eclipse/che/issues/1770)

### CODE REVIEW

**Comment by akurinnoy** [comment]
Workspace details:
workspace-details

**Comment by akurinnoy** [comment]
Project details:
project-details

**Comment by akurinnoy** [comment]
Project repository:
project-repository

**Comment by ashumilova** [comment]
Code of the PR : +1
@slemeur , please, take a look at UI
http://a10.codenvy-stg.com/ - for testing

**Comment by ashumilova** [comment]
Couple of questions for details:
1. Pencil is shown only for editable inputs or for all types (disabled for readonly input)
2.  Workspace info in projects details should be like readonly or disbaled state?
3. Tab titles should be always centered - same for the tabs in project creation flow?
@slemeur @TylerJewell

**Comment by ashumilova** [comment]
@akurinnoy please provide the latest version of creation page view

**Comment by akurinnoy** [comment]
Create project
create-project

**Comment by TylerJewell** [comment]
@ashumilova @akurinnoy.  Here are some notes after reviewing the items.

1: In the workspace details, the gap between "stop" and "snapshot" buttons seems unusually large. The same is there for "export as a file" and "export to private cloud".  Can we look at a more normalized space, please?

2: On the "New Project" view, can you update the light gray text under Template to read, "Select a code template to create a new project in the workspace."

Otherwise, A+ - love it.

**Comment by akurinnoy** [comment]
@TylerJewell 
Create project:
create-project
Workspace details:
workspace-details

**Comment by TylerJewell** [comment]
@akurinnoy - looks good on the improvements.

I just realized that we have some other issues:
1: Change "New Project" in title to be "New Workspace".

2: Question about overall direction - shouldn't this be the same page as Configure Workspace?  Shouldn't the "Create Workspace" and "Edit Workspace Settings" tab be the same view?  Or, must we keep them separate because a user cannot modify the template after ws created?

3: Change "Load Workspace and Create Project" text in buttons on top & bottom to be "Create".  This is more consistent with other text buttons.

4: It seems weird that the button on the bottom is blue.  It's the same color as the highlights.  Is blue the intended color for "accept" buttons like this?  I think the answer is yes - but just want to ask the question to see what everyone thinks.

**Comment by slemeur** [comment]
The overall work is nice, there are few details that are needing refinements and polish.

When user hits the button "+" from Dashboard view:
- [x] Rename "New Project" by "New Workspace"
- [x] Make the width of the "Create" button as wide as the form
  screen shot 2016-08-15 at 10 44 02
- [x] In the stack section, remove the small sentence before the stacks (in ready-to-go stacks and stack library)
  screen_shot_2016-08-15_at_10_44_20
- [x] Make the font used for "create" buttons consistent (between the top right button and the one at the bottom of the form).

In Factory Creation form:
- [ ] Remove “Github” tab
- [ ] Rename the title of the screen by “New Factory From"
- [ ] Rename tabs: “Workspaces” by “Workspace” / “Config file” by “Factory Config”

Also see https://github.com/codenvy/codenvy/pull/533

File: dashboard/src/app/projects/create-project/create-project.html [hunk 1]
```diff
@@ -85,15 +87,18 @@
...
+                         che-label-description="Choose where the initial source code for this project will come from. The choice of templates and samples will vary by the stack chosen.">
+      <md-radio-group ng-model="createProjectCtrl.selectSourceOption"
+                      ng-change="createProjectCtrl.onSourceOptionChanged()">
         <md-radio-button value="select-source-new">New from blank, template, or sample project</md-radio-button>
         <md-radio-button value="select-source-existing">Import from existing location</md-radio-button>
       </md-radio-group>
 
       <div ng-show="createProjectCtrl.selectSourceOption === 'select-source-existing'">
-        <md-tabs md-stretch-tabs="always" md-dynamic-height md-selected="createProjectCtrl.selectedTabIndex">
+        <md-tabs md-stretch-tabs="auot" md-dynamic-height md-selected="createProjectCtrl.selectedTabIndex">
```
Comment (ashumilova) [inline dashboard/src/app/projects/create-project/create-project.html]: md-stretch-tabs="auto"

File: dashboard/src/app/workspaces/workspace-details/workspace-details.controller.js [hunk 2]
```diff
@@ -96,36 +96,100 @@ export class WorkspaceDetailsCtrl {
...
+  /**
+   * Updates workspace info.
+   */
+  doUpdateWorkspace() {
     this.workspaceDetails = this.cheWorkspace.getWorkspacesById().get(this.workspaceId);
     let workspaceNewDetails = angular.copy(this.workspaceDetails);
+
     workspaceNewDetails.config.name = this.newName;
+
+    workspaceNewDetails.config.environments.forEach((env) => {
```
Comment (ashumilova) [inline dashboard/src/app/workspaces/workspace-details/workspace-details.controller.js]: lodash can be used here same as https://github.com/eclipse/che/pull/2061/files#diff-62c417fbca8214333da4620deda20be7R108

**Pull request merged**